core/field_colour_slider.js

```diff
@@ -45,343 +35,364 @@ goog.require('goog.ui.Slider');
-            this.activateEyedropperInternal_);
-  }
+    // Set value updates the slider positions
+    // Do this before attaching callbacks to avoid extra events from initial set
+    this.setValue(this.getValue());
 
-  Blockly.DropDownDiv.setColour('#ffffff', '#dddddd');
-  Blockly.DropDownDiv.setCategory(this.sourceBlock_.parentBlock_.getCategory());
-  Blockly.DropDownDiv.showPositionedByBlock(this, this.sourceBlock_);
-
-  // Set value updates the slider positions
-  // Do this before attaching callbacks to avoid extra events from initial set
-  this.setValue(this.getValue());
-
-  // Enable callbacks for the sliders
-  this.sliderCallbacksEnabled_ = true;
-
-  Blockly.FieldColourSlider.hueChangeEventKey_ = goog.events.listen(this.hueSlider_,
-      goog.ui.Component.EventType.CHANGE,
-      this.sliderCallbackFactory_('hue'));
-  Blockly.FieldColourSlider.saturationChangeEventKey_ = goog.events.listen(this.saturationSlider_,
-      goog.ui.Component.EventType.CHANGE,
-      this.sliderCallbackFactory_('saturation'));
-  Blockly.FieldColourSlider.brightnessChangeEventKey_ = goog.events.listen(this.brightnessSlider_,
-      goog.ui.Component.EventType.CHANGE,
-      this.sliderCallbackFactory_('brightness'));
-};
-
-Blockly.FieldColourSlider.prototype.dispose = function() {
-  if (Blockly.FieldColourSlider.hueChangeEventKey_) {
-    goog.events.unlistenByKey(Blockly.FieldColourSlider.hueChangeEventKey_);
-  }
-  if (Blockly.FieldColourSlider.saturationChangeEventKey_) {
-    goog.events.unlistenByKey(Blockly.FieldColourSlider.saturationChangeEventKey_);
+    // Enable callbacks for the sliders
+    this.sliderCallbacksEnabled_ = true;
+
+    this.hueChangeEventKey_ = Blockly.browserEvents.bind(
+        this.hueSlider_, 'input', this, this.sliderCallbackFactory_('hue'));
+    this.saturationChangeEventKey_ = Blockly.browserEvents.bind(
+        this.saturationSlider_, 'input', this, this.sliderCallbackFactory_('saturation'));
+    this.brightnessChangeEventKey_ = Blockly.browserEvents.bind(
+        this.brightnessSlider_, 'input', this, this.sliderCallbackFactory_('brightness'));
   }
-  if (Blockly.FieldColourSlider.brightnessChangeEventKey_) {
-    goog.events.unlistenByKey(Blockly.FieldColourSlider.brightnessChangeEventKey_);
+
+  dispose() {
+    if (this.hueChangeEventKey_) {
+      Blockly.browserEvents.unbind(this.hueChangeEventKey_);
+    }
+    if (this.saturationChangeEventKey_) {
+      Blockly.browserEvents.unbind(this.saturationChangeEventKey_);
+    }
+    if (this.brightnessChangeEventKey_) {
+      Blockly.browserEvents.unbind(this.brightnessChangeEventKey_);
+    }
+    if (this.eyedropperEventData_) {
+      Blockly.browserEvents.unbind(this.eyedropperEventData_);
+    }
+    Blockly.Events.setGroup(false);
+    super.dispose();
   }
-  if (Blockly.FieldColourSlider.eyedropperEventData_) {
-    Blockly.unbindEvent_(Blockly.FieldColourSlider.eyedropperEventData_);
+
+  // From Closure
```

Comment:
Since this is ours now could we make it better? in particular, return an obj instead of an array, and make brightness 0-100 instead of 0-255. We're having to convert all over the place and it's kind of confusing.

If that's an annoying about of effort also totally cool to skip :P

Reply:
Updated to return an object, but leaving brightness alone since [hsvToHex](https://github.com/google/blockly/blob/develop/core/utils/colour.ts#L150) expects it to be in the 0-255 range and we call that.